File: Seminar/src/main/java/org/sopt/Seminar/domain/user/controller/UserController.java
package org.sopt.Seminar.domain.user.controller;

import java.net.URI;
import lombok.RequiredArgsConstructor;
import org.sopt.Seminar.domain.user.dto.request.UserRequest;
import org.sopt.Seminar.domain.user.service.UserService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/users/")
public class UserController {

    private final UserService userService;

    @PostMapping("sign-up")
    public ResponseEntity<Void> signUp(@RequestBody UserRequest request) {
        URI location = URI.create(userService.create(request));
        return ResponseEntity.created(location).build();
    }

    @PostMapping("sign-in")
    public ResponseEntity<Void> signIn(@RequestBody UserRequest request) {
        userService.signIn(request);
        return ResponseEntity.noContent().build();
    }
}

File: kopring/week2/kotlin-semina/src/main/kotlin/org/sopt/kotlinsemina/domain/user/controller/UserController.kt
package org.sopt.kotlinsemina.domain.user.controller

import org.sopt.kotlinsemina.domain.user.dto.request.UserRequest
import org.sopt.kotlinsemina.domain.user.service.UserService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.net.URI

@RestController
@RequestMapping("/api/users/")
class UserController(
    private val userService: UserService
) {

    @PostMapping("sign-up")
    fun signUp(@RequestBody request: UserRequest): ResponseEntity<Void> {
        val location = URI.create(userService.create(request))
        return ResponseEntity.created(location).build()
    }

    @PostMapping("sign-in")
    fun signIn(@RequestBody request: UserRequest): ResponseEntity<Void> {
        userService.signIn(request)
        return ResponseEntity.noContent().build()
    }
}